uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1016.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1016 {
+
+  /**
+   * Before PCHR-3094, changing the Absence Type of a Leave Request to
+   * another one with a different calculation unit would result in the
+   * respective record in the database having non-mandatory fields with
+   * wrong values. For example, if it was changed from an Absence Type in
+   * days, to another one in hours, the from_date_type and to_date_type
+   * (which are required only for Leave Requests in days) would still
+   * keep the old values and this would cause validation issues that
+   * prevented the Leave Request from being updated.
+   *
+   * This upgrader searches for any Leave Request in that situation and
+   * sets the optional fields (according to the calculation unit) as NULL.
+   *
+   * @return bool
+   */
+  public function upgrade_1016() {
+    $calculationUnits = array_flip(AbsenceType::buildOptions('calculation_unit', 'validate'));
+
+    $leaveRequestTable = LeaveRequest::getTableName();
+    $absenceTypeTable = AbsenceType::getTableName();
+
+    $query = "
```

Comment:
Is it not possible to do this using the API?

Reply:
The API doesn't support complex conditions like the one used here